### PR TITLE
update action to Node 20 and bump packages

### PR DESCRIPTION
**Pull request by Djontleman**
## Problem

Node 16 actions are [deprecated](https://github.blog/changelog/2023-09-22-github-actions-transitioning-from-node-16-to-node-20/). This action should be updated to Node 20.

## Changes

- Update action and workflows to use Node 20
- Update dependencies (see comment)
- Remove unused files

## Testing

Tested in my own fork after merging the same changes (https://github.com/Djontleman/workflow-dispatch/pull/1).

Workflow run: https://github.com/Djontleman/workflow-dispatch/actions/runs/8785890431

### CODE REVIEW

File: index.js
Comment (Djontleman) [inline index.js:1]: Removed this file as is not used from what I can tell. May have been left over from the original action.

File: package-lock.json
Comment (Djontleman) [inline package-lock.json:1]: Removed as this version of the action appears to use `yarn`. Also left over from original action which used `npm`.

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "workflow-dispatch-and-wait",
-  "version": "3.0.0",
+  "version": "4.0.0",
```
Comment (Djontleman) [inline package.json:3]: Most actions have seen have treated going from Node 16 to Node 20 as a breaking change.

**Comment by Djontleman** [comment]
Hi @aurelien-baudet, would you be able to review?

**Comment by StummeJ** [comment]
@Djontleman given this repo looks abandoned, could you tag this in your repo as `v4`, `v4.0`, and `v4.0.0`?

**Comment by Djontleman** [comment]
I have created `v4`, `v4.0` and `v4.0.0` tags: https://github.com/Djontleman/workflow-dispatch/releases/tag/v4.0.0
I haven't published to Marketplace though, not sure on the etiquette here.

**Comment by aurelien-baudet** [comment]
Sorry, I have no more time and energy to maintain this repository. I even don't have computer anymore. Someone of you should for this repository and merge the different pull requests.
Tell me who creates the fork and I will archive this repository but indicate which repository follows up

**Comment by fbiesse** [comment]
@Djontleman I wanted to know if you're planning to take over on this action.
If not, I propose to create a fork, update readme, integrate your developments and maintain it.
Tell me

**Comment by VincentLanglet** [comment]
Hi, FYI https://github.com/benc-uk/workflow-dispatch/issues/73

**Pull request merged**